File: Modules/ServerTypes/OrganisationHierarchy/DivisionForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface DivisionForm {
    Name: StringEditor;
    Description: StringEditor;
}

export class DivisionForm extends PrefixedContext {
    static formKey = 'OrganisationHierarchy.Division';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DivisionForm.init)  {
            DivisionForm.init = true;

            var w0 = StringEditor;

            initFormType(DivisionForm, [
                'Name', w0,
                'Description', w0
            ]);
        }
    }
}

File: Modules/HumanResource/OrganisationHierarchy/Division/DivisionForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.OrganisationHierarchy.Forms;

[FormScript("OrganisationHierarchy.Division")]
[BasedOnRow(typeof(DivisionRow), CheckNames = true)]
public class DivisionForm
{
    public string Name { get; set; }
    public string Description { get; set; }
}